### PR TITLE
Add @nogc attribute to several BigInt accessor functions

### PR DESCRIPTION
**Pull request by JakobOvrum**

### CODE REVIEW

File: std/internal/math/biguintcore.d [hunk 1]
```diff
@@ -82,10 +82,10 @@ template maxBigDigits(T) if (isIntegral!T)
...
 }
 
-enum immutable(BigDigit) [] ZERO = [0];
-enum immutable(BigDigit) [] ONE = [1];
-enum immutable(BigDigit) [] TWO = [2];
-enum immutable(BigDigit) [] TEN = [10];
+static immutable BigDigit[] ZERO = [0];
+static immutable BigDigit[] ONE = [1];
+static immutable BigDigit[] TWO = [2];
+static immutable BigDigit[] TEN = [10];
```
Comment (JakobOvrum) [inline std/internal/math/biguintcore.d]: Note: this avoids GC-allocation upon use

**Comment by andralex** [comment]
thx!

File: std/internal/math/biguintcore.d [hunk 2]
```diff
@@ -108,12 +108,12 @@ private:
         opAssign(x);
     }
 
-    enum trustedAssumeUnique = function(BigDigit[] input) pure @trusted {
+    enum trustedAssumeUnique = function(BigDigit[] input) pure @trusted @nogc {
         return assumeUnique(input);
```
Comment (quickfur) [inline std/internal/math/biguintcore.d]: Isn't this an abuse of `@trusted`, per the recent forum discussions? We are relying on the caller to make sure that the argument is, in fact, unique, but what if it isn't?

Comment (yebblies) [inline std/internal/math/biguintcore.d]: Yes.

**Pull request merged**